src/index.test.ts

```diff
@@ -1114,6 +1114,79 @@ test('Should call getVariant event when impressionData is true', (done) => {
     client.on(EVENTS.IMPRESSION, (event: any) => {
         expect(event.featureName).toBe('impression-variant');
         expect(event.eventType).toBe('getVariant');
+        expect(event.impressionData).toBe(true);
+        client.stop();
+        done();
+    });
+});
+
+test('Should not call isEnabled event when impressionData is false', (done) => {
+    const bootstrap = [
+        {
+            name: 'impression',
+            enabled: true,
+            variant: {
+                name: 'disabled',
+                enabled: false,
+            },
+            impressionData: false,
+        },
+    ];
+
+    const config: IConfig = {
+        url: 'http://localhost/test',
+        clientKey: '12',
+        appName: 'web',
+        bootstrap,
+    };
+    const client = new UnleashClient(config);
+    client.start();
+
+    client.on(EVENTS.READY, () => {
+        const isEnabled = client.isEnabled('impression');
+        expect(isEnabled).toBe(true);
+        client.stop();
+        done();
+    });
+
+    client.on(EVENTS.IMPRESSION, () => {
+        client.stop();
+        fail('SDK should not emit impression event');
+    });
+});
+
+test('Should call getVariant event when impressionData is false and impressionDataAll is true', (done) => {
+    const bootstrap = [
+        {
+            name: 'impression-variant',
+            enabled: true,
+            variant: {
+                name: 'disabled',
+                enabled: false,
+            },
+            impressionData: false,
+        },
+    ];
+
+    const config: IConfig = {
+        url: 'http://localhost/test',
+        clientKey: '12',
+        appName: 'web',
+        bootstrap,
+        impressionDataAll: true,
+    };
+    const client = new UnleashClient(config);
+    client.start();
+
+    client.on(EVENTS.READY, () => {
+        const isEnabled = client.getVariant('impression-variant');
+        expect(isEnabled).toBe(true);
+    });
```

Comment:
There's no test for whether impressionData events are returned when a toggle is `false` but `impressionDataAll` is true: is that supposed to have the same behavior?

Reply:
yes, I should add a test case for this as well. I did it for getVariant.